examples/amm/src/state.rs

```diff
@@ -1,11 +1,27 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use linera_sdk::views::{linera_views, RegisterView, RootView, ViewStorageContext};
+use fungible::Account;
+use linera_sdk::{
+    base::Amount,
+    views::{linera_views, MapView, RegisterView, RootView, ViewStorageContext},
+};
 
 #[derive(RootView, async_graphql::SimpleObject)]
 #[view(context = "ViewStorageContext")]
 pub struct Amm {
-    // TODO(#968): We should support stateless Applications/empty user views
-    pub _dummy: RegisterView<u8>,
+    pub shares: MapView<Account, Amount>,
+    pub total_shares_supply: RegisterView<Amount>,
+}
+
+#[allow(dead_code)]
```

Comment:
If it's only used in the contract, you could move this `impl` block into `contract.rs` instead.